### ISSUE TITLE
[FEATURE REQUEST] keep old branches?

### ISSUE THREAD

**Issue by emmanueltouzery** [NTBBloodbath/doom-nvim#298]
Currently doom-nvim has three branches in its git repo: main, develop, and next.
What I think would be useful is to keep branches around for old stable releases.

So the current branches would stay, but new branches would be added: v3.1.x, v3.2.x, and so on, as releases are being added. It could be useful for people who want/need to keep using old nvim releases for instance, or if there are changes in newer versions that they don't like.

**Comment by NTBBloodbath** [comment]
Hi, thanks for this idea!

I don't think this is really needed and I know it's not exactly the same thing but we store previous _and current_ stable releases in our [releases](https://github.com/NTBBloodbath/doom-nvim/releases) page.

_See explanation below to know why I don't think it's needed._

---

> It could be useful for people who want/need to keep using old nvim releases for instance, or if there are changes in newer versions that they don't like.

I'll explain you how does Doom works internally when updating or rolling back to a previous version just to clarify some things. Our updating and rollback systems are very complex internally but they're actually very easy to explain. Basically Doom creates _or regenerates_ a hidden file in your Neovim configuration path called `.doom_releases` when updating itself (`:DoomUpdate`) that file contains all the information about our releases tags, e.g. `v3.2.0` and that version commit sha.

This file is useful for us to also rollback to a previous version with our rollback system (`:DoomRollback`), so let's say you're currently in `v3.2.0` but you want to go back to `v3.1.2`. You can easily go back with our rollback command that will check our `.doom_releases` file and compare all versions until it finds a lower version.

This algorithm is a bit different for rolling (aka develop) branch users, it does use a different file name and rollbacks to a previous commit instead of rolling back to a previous Doom release.

So, basically what you want is already implemented in Doom. However, in case you want to manually rollback to a previous version you should be able to use the following Git command.

```sh
git checkout tags/tag_name # e.g. v3.2.0
```

I don't personally recommend using old versions because we are pinning our plugins commits now and there can be conflicts with your plugins if you're using an old Neovim version like `0.5.x` and also if Neovim suffers internal breaking changes in the Lua API in the future. However you're free to do it if you want, just wanted to advice you that this can _or not_ carry to some future problems.

Hope all my text didn't bother you haha just wanted to let you know some things in a detailed way.

Cheers

**Comment by emmanueltouzery** [comment]
Yes, with a checkout I'm basically good to go. It's more that I don't have a backup online unless I clone the repo. But it's ok, I understand, especially if you consider running old versions a bad idea.

**Comment by NTBBloodbath** [comment]
Hmm cloning the repo isn't really needed, unless you want to have all the commits history and such things. You can also download the release source code (`.zip` or `.tar.gz`) if you want to have a static doom installation, this source code includes everything but the `.git` directory.

**Comment by emmanueltouzery** [comment]
Oh that is true! The source code from the tagging on GitHub. It seems to keep working despite the force pushes. That would be the fix for me then, yes! (I'm sticking to 0.5 for now due to a regression with elixir in 0.6). Thank you!
